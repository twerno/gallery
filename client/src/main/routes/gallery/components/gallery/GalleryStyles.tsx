import { breakpoints } from 'main/globalStyles';
import styled, { css } from 'styled-components';

export const GalleryContainer = styled.div`
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    overflow-x: hidden;
    position: relative;
    left: 0px;
    right: 0px;
    margin: 0px 10px;

    &::after {
        content: '';
        flex-grow: 999999999;
        height: 0;
    }
`;

export const GalleryItemPlaceholderWraper = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 500px;
  width: 100%;
  height: 100%;

  @media only screen and (max-width: ${breakpoints.md}) {

    & {
      min-width: 200px;
    }
  }
`;


export interface IGalleryItemProps {
  disabled?: boolean;
}

export const GalleryItem = styled.div<IGalleryItemProps>`
  margin: 1px;
  flex-grow: 1;
  overflow: hidden;
  display: flex;
  height: 200px;

  @media only screen and (max-width: ${breakpoints.md}) {

      & {
          height: 100px;
      }
  }

  pointer-events: ${props => props.disabled ? 'none' : 'initial'};
`;

/**
* 
*/
interface IGalleryPicture {
  readonly isHidden: boolean;
}

/**
* 
*/
export const GalleryPicture = styled.picture<IGalleryPicture>`
  opacity: 1;
  position: relative;
  transition: opacity 0.2s ease-in-out;
  min-width: 100%;

  ${
  props => props.isHidden && css`
          position: absolute;
          opacity: 0;
  `}
`;

export const GalleryImg = styled.img`
  object-fit: cover;
  min-width: 100%;
  height: 100%;
`;