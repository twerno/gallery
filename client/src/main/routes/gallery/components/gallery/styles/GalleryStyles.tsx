import { breakpoints } from 'main/globalStyles';
import styled, { css } from 'styled-components';

const GalleryContainer = styled.div`
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    overflow-x: hidden;
    position: relative;
    left: 0px;
    right: 0px;
    margin: 0px 10px;

    &::after {
        content: '';
        flex-grow: 999999999;
        height: 0;
    }
`;

const ItemLoadingPlaceholder = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 500px;
  width: 100%;
  height: 100%;

  @media only screen and (max-width: ${breakpoints.md}) {

    & {
      min-width: 200px;
    }
  }
`;


export interface IGalleryStyledItemProps {
  disabled?: boolean;
}

const GalleryItem = styled.div<IGalleryStyledItemProps>`
  margin: 1px;
  flex-grow: 1;
  overflow: hidden;
  display: flex;
  height: 200px;

  @media only screen and (max-width: ${breakpoints.md}) {

      & {
          height: 100px;
      }
  }

  pointer-events: ${props => props.disabled ? 'none' : 'initial'};
`;

/**
* 
*/
interface IGalleryStyledPictureProps {
  readonly visible: boolean;
}

/**
* 
*/
const GalleryItemPicture = styled.picture<IGalleryStyledPictureProps>`
  opacity: 1;
  position: relative;
  transition: opacity 0.2s ease-in-out;
  min-width: 100%;

  ${
  props => !props.visible && css`
          position: absolute;
          opacity: 0;
  `}
`;

const GalleryItemImg = styled.img`
  object-fit: cover;
  min-width: 100%;
  height: 100%;
`;


export default {
  GalleryContainer,
  ItemLoadingPlaceholder,
  GalleryItem,
  GalleryItemPicture,
  GalleryItemImg,
}