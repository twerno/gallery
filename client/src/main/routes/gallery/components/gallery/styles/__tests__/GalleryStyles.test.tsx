import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as React from 'react';
import { componentBasicTests } from 'test/componentBasicTests';
import GalleryStyles from '../GalleryStyles';

describe('GalleryStyles', () => {

    componentBasicTests(
        "GalleryContainer",
        <GalleryStyles.GalleryContainer />
    );

    componentBasicTests(
        "GalleryItem - disabled=true",
        <GalleryStyles.GalleryItem disabled={true} />
    );

    componentBasicTests(
        "GalleryItem - disabled=false",
        <GalleryStyles.GalleryItem disabled={false} />
    );

    componentBasicTests(
        "GalleryItem - disabled:default",
        <GalleryStyles.GalleryItem />
    );

    componentBasicTests(
        "GalleryItemImg",
        <GalleryStyles.GalleryItemImg />
    );

    componentBasicTests(
        "GalleryItemPicture - visible=true",
        <GalleryStyles.GalleryItemPicture visible={true} />
    );

    componentBasicTests(
        "GalleryItemPicture - visible=false",
        <GalleryStyles.GalleryItemPicture visible={false} />
    );

    componentBasicTests(
        "ItemLoadingPlaceholder",
        <GalleryStyles.ItemLoadingPlaceholder />
    );

});

