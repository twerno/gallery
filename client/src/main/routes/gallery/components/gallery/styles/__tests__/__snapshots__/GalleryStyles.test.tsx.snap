// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryStyles [GalleryContainer] Component basic tests matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  overflow-x: hidden;
  position: relative;
  left: 0px;
  right: 0px;
  margin: 0px 10px;
}

.c0::after {
  content: '';
  -webkit-box-flex: 999999999;
  -webkit-flex-grow: 999999999;
  -ms-flex-positive: 999999999;
  flex-grow: 999999999;
  height: 0;
}

<div
  className="c0"
/>
`;

exports[`GalleryStyles [GalleryItem - disabled:default] Component basic tests matches the snapshot 1`] = `
.c0 {
  margin: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 200px;
  pointer-events: initial;
}

@media only screen and (max-width:600px) {
  .c0 {
    height: 100px;
  }
}

<div
  className="c0"
/>
`;

exports[`GalleryStyles [GalleryItem - disabled=false] Component basic tests matches the snapshot 1`] = `
.c0 {
  margin: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 200px;
  pointer-events: initial;
}

@media only screen and (max-width:600px) {
  .c0 {
    height: 100px;
  }
}

<div
  className="c0"
  disabled={false}
/>
`;

exports[`GalleryStyles [GalleryItem - disabled=true] Component basic tests matches the snapshot 1`] = `
.c0 {
  margin: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 200px;
  pointer-events: none;
}

@media only screen and (max-width:600px) {
  .c0 {
    height: 100px;
  }
}

<div
  className="c0"
  disabled={true}
/>
`;

exports[`GalleryStyles [GalleryItemImg] Component basic tests matches the snapshot 1`] = `
.c0 {
  object-fit: cover;
  min-width: 100%;
  height: 100%;
}

<img
  className="c0"
/>
`;

exports[`GalleryStyles [GalleryItemPicture - visible=false] Component basic tests matches the snapshot 1`] = `
.c0 {
  opacity: 1;
  position: relative;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
  min-width: 100%;
  position: absolute;
  opacity: 0;
}

<picture
  className="c0"
/>
`;

exports[`GalleryStyles [GalleryItemPicture - visible=true] Component basic tests matches the snapshot 1`] = `
.c0 {
  opacity: 1;
  position: relative;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
  min-width: 100%;
}

<picture
  className="c0"
/>
`;

exports[`GalleryStyles [ItemLoadingPlaceholder] Component basic tests matches the snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 500px;
  width: 100%;
  height: 100%;
}

@media only screen and (max-width:600px) {
  .c0 {
    min-width: 200px;
  }
}

<div
  className="c0"
/>
`;
